service/posts/views.py

```diff
@@ -11,20 +11,12 @@ class PublicPostsList(generics.ListAPIView):
     pagination_class = PostsPagination
     serializer_class = PostSerializer
     authentication_classes = (NodeBasicAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    # No permission class
 
     def get_queryset(self):
-        node = self.request.user
-
-        if not node.share_posts:
-            return Post.objects.none()
-
-        queryset = Post.objects.filter(visibility="PUBLIC")
+        remote_node = self.request.user
```

Comment:
Oh, I do recall that in our code. That's interesting! If I'm correct, there's only a potential for problems with this code if `authentication_classes` is not enforced to use `NodeBasicAuthentication`? For example, if plain session authentication is used (e.g. by some `Author` who's signed in). In this PR I can see that it's enforced and my question is more just for understanding potential edge cases in case of future modifications to this file or the other service classes.